src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+
+        @Argument(
+                doc = "Output file with only the configuration settings in it.",
+                fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+                shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME
+        )
+        private File outputFilePath;
+
+        @Override
+        public void traverse() {
+            final GATKConfig gatkConfig = ConfigFactory.getInstance().getOrCreate(GATKConfig.class);
+            ConfigFactory.dumpConfigSettings( gatkConfig, outputFilePath.toPath() );
+        }
+    }
+
+    @Override
+    public String getTestedClassName() {
+        return DummyGatkTool.class.getSimpleName();
+    }
+
+    @DataProvider
+    private Object[][] provideForTestToolWithConfigValidation() {
+        return new Object[][] {
+                { getSafeNonExistentPath("nonExistentConfig.config") },
+        };
+    }
+
+    @Test(dataProvider = "provideForTestToolWithConfigValidation")
+    public void testToolWithConfigValidation(final Path configFilePath) {
+        // From droazen:
+//      A better integration test, I think, would be to create a private static dummy tool
+//      class here that just dumps selected items from its configuration to a file in apply(), run that dummy tool
+//      in your test, and assert that the configuration dumped is consistent with the one specified on the command line.
+//
+//      You could have two test cases: one with the default config file, and one with a user-specified config file
+//      that alters one or more default settings.
+
+        final File tmpFile = getSafeNonExistentFile("ConfigIntegrationTest.output.txt");
+        tmpFile.deleteOnExit();
+
+        // Create some arguments for our command:
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
+        args.add(configFilePath.toUri().toString());
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(tmpFile);
+
+        // Run our command:
+        runCommandLine(args.getArgsArray());
+
+        // Now we get to read in the file's contents and compare them to our config settings:
+        final Properties properties = new Properties();
+        try ( final FileInputStream inputStream = new FileInputStream(tmpFile) ) {
+            properties.load(inputStream);
+        }
+        catch (final Exception ex) {
+            throw new GATKException("Test error!", ex);
+        }
+
+        // Create a config object to compare with the properties:
+        final GATKConfig config = ConfigFactory.getInstance().getGATKConfig();
+
+        final LinkedHashMap<String, Object> configMap = ConfigFactory.getConfigMap(config, false);
+
+        // Need to use reflection here to make this test future-proof.
+        // Now we cycle through our interface methods, resolve parameter names,
+        // and assert their equality with the properties we have read:
```

Comment:
Are you actually asserting equality with the properties read in from the `DummyGatkTool` output? I don't see where...

Reply:
Nope.  I ended up comparing the wrong sets of properties for this test.  Fixing now.